app/src/main/java/ir/jaamebaade/jaamebaade_client/viewmodel/SearchViewModel.kt

```diff
@@ -69,10 +99,46 @@ class SearchViewModel @Inject constructor(
         }
     }
 
+    private suspend fun saveSearchHistory(searchQuery: String) {
+        val searchHistory = SearchHistoryRecord(
+            query = searchQuery,
+            timestamp = System.currentTimeMillis()
+        )
+        searchHistoryRepository.insertSearchHistory(searchHistory)
+    }
+
     private suspend fun getAllPoets() {
         withContext(Dispatchers.IO) {
             _allPoets.value = poetRepository.getAllPoets()
         }
 
     }
+
+    fun deleteHistoryItem(historyItem: SearchHistoryRecord) {
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
+                searchHistoryRepository.removeHistoryRecord(historyItem)
+            }
+
+        }
+
+    }
+
+    fun onQueryChanged() {
```

Comment:
remove comments. also, move public functions to top of class, just below other public functions